### PR TITLE
Records with different terms and tabs

### PR DESCRIPTION
**Pull request by IlievaDayana**

### CODE REVIEW

File: src/components/Stream/assistant.tsx [hunk 1]
```diff
@@ -57,7 +58,7 @@ const Assistant = ({ message, itemId }: AssistantProps) => {
             <Media
               e2e="assistant-iframe"
               key={uuidV4()}
-              title={it[it.type]?.title||''}
+              title={it[it.type]?.title || ''}
               background={`url(https://img.youtube.com/vi/${extractVideoCode(it[it.type]?.url)}/hqdefault.jpg)`}
               url={extractVideoCode(it[it.type]?.url || 'https://www.youtube.com/embed/g4B8Dhl4pxY')}
               type={Definition.video}
```
Comment (venzovisa) [inline src/components/Stream/assistant.tsx:64]: @IlievaDayana I don't see any fallback for the background similar to the one for `url` attribute. Can you provide something?

File: src/middleware/socket.ts [hunk 2]
```diff
@@ -142,6 +143,7 @@ const chatMiddleware: Middleware = (store) => (next) => (action) => {
   }
 
   if (resendMessage.match(action) && !('$isSync' in action)) {
+    // @ts-ignore must understand why the action is of type never??
     handleMessageResending(action.payload);
   }
```
Comment (venzovisa) [inline src/middleware/socket.ts:148]: @IlievaDayana What happens if you check existence of `payload` property in `action` before use and what is actual type of `Action` from Redux toolkit?

Comment (IlievaDayana) [reply src/middleware/socket.ts:148]: i've tried adding type check also property check and both also, but it did not work?
Action<T> where T stands for template, maybe i should describe it somehow.

**Review by venzovisa — APPROVED** [review]

File: __tests__/feature/chat-streaming.spec.jsx [hunk 3]
```diff
@@ -148,42 +220,65 @@ describe('AppBase, streaming events execute properly', () => {
...
+        {
+          id: '4216d0a8-2621-499c-8a65-70191e31ee5a',
+          type: 'payment',
+          sequence: 1,
+          payment: 'provide payment',
+          term,
+          threadId: threadId,
+        },
+      ]);
     });
```
Comment (venzovisa) [inline __tests__/feature/chat-streaming.spec.jsx:283]: @IlievaDayana I see a lot of duplicate objects. You can create factory function that produces mocked objects and call it wherever needed:

const generateStreamingData = (data: Partial<StreamingData>) => ({
          type: 'text',
          text: '',
          sequence: 1,
          id: '4216d0a8-2621-499c-8a65-70191e31ee5a',
          term: '',
          threadId: faker.random.uuid(),
          ...data
});

File: __tests__/feature/special-messaging.spec.jsx [hunk 4]
```diff
@@ -24,37 +42,50 @@ describe('Special messages are hidden and elements depending on them are visuali
...
-        "history": [],
-        "errors": []
-      })
-      jest.advanceTimersByTime((initialMessage.length * 1000) * initialMessage.length - 1);
+        region: faker.location.country(),
+        history: [],
+        errors: [],
+        term: term,
+        threadId: threadId,
+      });
```
Comment (venzovisa) [inline __tests__/feature/special-messaging.spec.jsx:57]: @IlievaDayana you can pass `term` and `threadId` only once for short version.

```
{
...
 term,
 threadId
}
```

File: src/store/slices/chat.ts [hunk 5]
```diff
@@ -197,22 +235,51 @@ const configSlice = createSlice({
...
+  };
+};
+
+export const sortBySequence = (a: MessageProperties, b: MessageProperties) => {
+  if (a.sequence && b.sequence) {
+    return a.sequence - b.sequence;
+  }
+
+  return 0;
+};
```
Comment (venzovisa) [inline src/store/slices/chat.ts:278]: @IlievaDayana Do you plan to add tests for these methods?

File: src/components/Stream/bubble.tsx [hunk 6]
```diff
@@ -23,7 +23,17 @@ const StreamBubble = ({ itemId }: { itemId: string }) => {
...
+        [...record.content].map((record) =>
+          <User
+            key={uuidV4()}
+            record={{
+              text: record.text || '',
+              resend: record.resend || false,
+              sent: record.sent || true,
+              groupId: record.groupId || '',
+              itemId: itemId
+            }}
```
Comment (venzovisa) [inline src/components/Stream/bubble.tsx:35]: @IlievaDayana could be passed only once:

```
{
    ...
    itemId
}
```

**Review by venzovisa — APPROVED** [review]

**Pull request merged**